Client/Ecosia/UI/Coordinators/MainCoordinator.swift

```diff
@@ -0,0 +1,43 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Core
+
+final class MainCoordinator: Coordinator {
+
+    var childCoordinators = [Coordinator]()
+    var navigationController: UINavigationController
+    var browserViewController: BrowserViewController!
+    weak var welcomeDelegate: WelcomeDelegate?
+    
+    private var profile: Profile
+    private var tabManager: TabManager
+
+    init(navigationController: UINavigationController,
+         profile: Profile,
+         tabManager: TabManager,
+         welcomeDelegate: WelcomeDelegate?) {
+        self.navigationController = navigationController
+        self.profile = profile
+        self.tabManager = tabManager
+        self.welcomeDelegate = welcomeDelegate
```

Comment:
I wonder if it makes sense to move the `welcomeDelegate` inside the MainCoordinator since it basically just handles the `navigationController`. This would make it possible to move the `navigationController` initialisation code in here as well as make it clearly responsible for everything related to the initial flow of the app 🤔

Reply:
That’s what I did in the first place yeah 🤣. Although I thought that the Coordinators shouldn’t have any other “behaviour” assigned to them implicitly.